analyses/infercnv-consensus-cell-type/template-notebooks/ewings-infercnv-results.Rmd

```diff
@@ -0,0 +1,536 @@
+To see whether we'd like to further tune the clustering parameters, we'll get a sense of their size and rough composition here.
+
+First, how many cells are there per cluster?
+Clusters labeled `reference_` contain only normal reference cells, and clusters labeled `unknown_` contain the query cells.
+
+```{r, fig.width = 10}
+ggplot(has_cnv_df) +
+  # keep in default order so "reference" and "unknown" are grouped
+  aes(x = subcluster) +
+  geom_bar() +
+  facet_wrap(
+    vars(normal_reference),
+    scales = "free_x"
+  ) +
+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
+```
+
+Let's also look at the distributions of cell types in clusters.
+Is there a trend for which cell types tend to show up in which cluster?
+We'll plot this for both the consensus and ewing annotations, considering only the top 7 cell types for visual ease.
+
+```{r fig.width=12}
+ggplot(query_has_cnv_df) +
+  aes(x = subcluster, fill = forcats::fct_lump(consensus_annotation, 7)) +
+  geom_bar() +
+  scale_fill_brewer(palette = "Dark2") +
+  facet_wrap(
+    vars(normal_reference),
+    scales = "free_x"
+  ) +
+  labs(
+    title = "Consensus cell types",
+    fill = "annotation"
+  ) +
+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
+```
+
+```{r fig.width=12}
+ggplot(query_has_cnv_df) +
+  aes(x = subcluster, fill = forcats::fct_lump(ewing_annotation, 7)) +
+  geom_bar() +
+  scale_fill_brewer(palette = "Set1") +
+  facet_wrap(
+    vars(normal_reference),
+    scales = "free_x"
+  ) +
+  labs(
+    title = "Ewing cell types",
+    fill = "annotation"
+  ) +
+  theme(axis.text.x = element_text(angle = 45, hjust = 1))
+```
+
+Let's also look more specifically at the `reference` clusters, where we might expect some batch effects since cells were taken from several different libraries.
+Do originating libraries cluster together?
+
+```{r fig.width=10}
+all_reference_clusters <- has_cnv_df |>
+  dplyr::filter(
+    normal_reference == "all-immune",
+    immune_reference_cell
+  )
+all_plot <- ggplot(all_reference_clusters) +
+  aes(x = subcluster, fill = library_id) +
+  geom_bar() +
+  labs("All immune reference") +
+  theme(legend.position = "bottom")
+
+
+subset_reference_clusters <- has_cnv_df |>
+  # bring in library_id
+  dplyr::inner_join(
+    infercnv_df |>
+      dplyr::select(barcodes, library_id) |>
+      unique()
+  ) |>
+  dplyr::filter(
+    normal_reference == "subset-immune",
+    subset_reference_cell
+  )
+subset_plot <- ggplot(subset_reference_clusters) +
+  aes(x = subcluster, fill = library_id) +
+  geom_bar() +
+  labs("Subset immune reference") +
+  theme(legend.position = "bottom")
+
+wrap_plots(all_plot, subset_plot, guides = "collect") &
+  theme(legend.position = "bottom")
+```
+
+
+How many CNVs were inferred per cluster?
+We expect a single value here, because with `analysis_mode = "subclusters"` (the `inferCNV` default), CNV parameters are estimated per cluster.
+
+```{r}
+# How many CNV per cell per cluster?
```

Comment:
I think this plot is really helpful! It looks like `unknown` clusters dominated by tumor cells have higher total CNV than clusters dominated by normal cells, which is a good thing! I think we might be able to use this information to help classify tumor cell clusters.